docs/architecture/adr-010-posts-module.md

```diff
@@ -0,0 +1,558 @@
+  required uint32 poll_id = 3;
+  
+  // Indexes of the answer inside the ProvidedAnswers array
+  repeated uint32 answers_indexes = 4;
+  
+  // Address of the user answering the poll
+  required string signer = 5;
+}
+
+// MsgAnswerPollResponse represents the MSg/AnswerPoll response type
+message MsgAnswerPollResponse {}
+```
+
+### `Query` Service
+```protobuf
+// Query defines the gRPC querier service
+service Query {
+  // Posts queries all the posts inside a given subspace
+  rpc Posts(QueryPostsRequest) returns (QueryPostsResponse) {
+    option (google.api.http).get = "/desmos/posts/v1/{subspace_id}/posts";
+  }
+  
+  // PostAttachments queries the attachments of the post having the given id
+  rpc PostAttachments(QueryPostAttachmentsRequest) returns (QueryPostAttachmentsResponse) {
+    option (google.api.http).get = "/desmos/posts/v1/{subspace_id}/posts/{post_id}/attachments";
+  }
+  
+  // PollAnswers queries the answers for the poll having the given id
+  rpc PollAnswers(QueryPollAnswersRequest) returns (QueryPollAnswersResponse) {
+    option (google.api.http).get = "/desmos/posts/v1/{subspace_id}/posts/{post_id}/polls/{poll_id}/answers";
+  }
+  
+  // PollTallyResults queries the tally results for an ended poll
+  rpc PollTallyResults(QueryPollTallyResultRequest) returns (QueryPollTallyResultResponse) {
+    option (google.api.http).get = "/desmos/posts/v1/{subspace_id}/posts/{post_id}/polls/{poll_id}/results";
+  }
+  
+  // Params queries the module parameters
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/desmos/posts/v1/params";
+  }
+}
+
+// QueryPostsRequest is the request type for the Query/Posts RPC method
+message QueryPostsRequest {
+  // Id of the subspace to query the posts for
+  required uint64 subspace_id = 1;
+  
+  // pagination defines an optional pagination for the request.
+  optional cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+// QueryPostsResponse is the response type for the Query/Posts RPC method
+message QueryPostsResponse {
+  repeated Post posts = 1;
+  required cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryPostsRequest is the request type for the Query/PostAttachments RPC method
+message QueryPostAttachmentsRequest {
+  // Id of the subspace where the post is stored 
+  required uint64 subspace_id = 1;
+  
+  // Id of the post to query the attachments for
+  required uint64 post_id = 2;
+
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 3;
+}
+
+// QueryPostAttachmentsResponse is the response type for the Query/PostAttachments RPC method
+message QueryPostAttachmentsResponse {
+  repeated Attachment attachments = 1;
+  required cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryPollAnswersRequest is the request type for the Query/PollAnswers RPC method
+message QueryPollAnswersRequest {
```

Comment:
To make things simpler, I've added a `user` field inside this request. It allows to specify a single user instead of multiple ones so that pagination can still be supported properly (allowing a repeated field would have caused some problems with it). If a developer wants to get multiple users' responses then they can call this query multiple times very easily